### PR TITLE
Improvement - "Adding new features for table component"

### PR DESCRIPTION
**Pull request by gustavookamoto**
While I was implementing the table component, I realise some improvements that would be good to have it:
- Hover optinon with event emitter (as edit and delete options)
- Clear fields with "cross-icon" inside inputs.
- Code review

image

fixes: #575

### CODE REVIEW

**Comment by helloimela** [comment]
Fix #576

**Comment by helloimela** [comment]
Hi, thanks for the PR!
I got this error when trying out this solution. When I typed on input, it does not do the filter and this error appears. 
Do you know why this happen?
image
image

File: src/components/table/table.tsx [hunk 1]
```diff
@@ -174,6 +182,19 @@ export class TableComponent {
...
 
+  private clearInputField(key) {
+    (document.getElementById("search" + key) as HTMLInputElement).value = "";
+    this.removeSortActive();
+    this.searchColumn();
+  }
+
+  /* Remove all sort-active class */
+  private removeSortActive() {
+    let actives = document.querySelectorAll(".sort-active");
```
Comment (helloimela) [inline src/components/table/table.tsx:193]: This should also use shadow root selector

**Review by helloimela — CHANGES_REQUESTED** [review]

**Comment by mmexvr** [comment]
@gustavookamoto FYI

**Review by helloimela — APPROVED** [review]
Looks good!

**Pull request merged**